docs/README.md

```diff
@@ -0,0 +1,14 @@
+# Documentation
+
+Welcome to the documentation for the **Hetzner Cloud CLI**.
+
+This documentation is written and organized following the [Diátaxis](https://diataxis.fr/) guidelines. Below you can find a high-level overview of where to look for certain things:
+
+- [Tutorials](tutorials)
+- [Guides](guides)
+- [Reference](reference)
```

Comment:
Unless we have some other reference guides, what do you think about linking to `reference/hcloud.md` directly?